mesonbuild/backend/ninjabackend.py

```diff
@@ -2997,7 +2997,7 @@ def generate_ending(self):
         elem = NinjaBuildElement(self.all_outputs, deps, 'phony', '')
         self.add_build(elem)
 
-    def get_introspection_data(self, target_id, target):
+    def get_introspection_data(self, target_id: str, target: build.Target) -> T.List[T.Dict[str, T.Union[bool, str, T.List[T.Union[str, T.Dict[str, T.Union[str, T.List[str], bool]]]]]]]:
```

Comment:
This is a place that without `TypedDict`, I'd be happy to use `T.Any` as the dict value. This sort of stuff is exactly what `TypedDict` is for.

Reply:
There is https://github.com/python/typing/tree/master/typing_extensions which could give us both `Protocol` and `TypedDict` for 3.6+

Reply:
I think that might be a good idea. If we always guard typing_extensions behind `if T.TYPE_CHECKING` then it would only be a requirement when using mypy.